packages/compiler/src/render3/r3_factory.ts

```diff
@@ -141,6 +141,13 @@ export interface R3DependencyMetadata {
    */
   token: o.Expression;
 
+  /**
+   * If an @Attribute decorator is present, this is the literal type of the attribute name, or
+   * the unknown type if no literal type is available (e.g. the attribute name is an expression).
+   * Will be null otherwise.
+   */
+  type: o.Expression|null;
```

Comment:
Shouldn't this be:
```suggestion
  attribute: o.Expression|null;
```
or
```suggestion
  attribute: string|null;
```

I think we should just pass the attribute name as a `string` here.